tools/ansible/tasks/wsu/test/e2e/wsu_test.go

```diff
@@ -134,5 +145,37 @@ func testFilesCopied(t *testing.T) {
 		assert.NoError(t, err, "Error looking for %s: %s", fullPath, err)
 		assert.Emptyf(t, stdout.String(), "Missing file: %s", fullPath)
 	}
+}
 
+// testNodeReady tests that the bootstrapped node was added to the cluster and is in the ready state
+func testNodeReady(t *testing.T) {
+	var createdNode *v1.Node
+	nodes, err := k8sclientset.CoreV1().Nodes().List(metav1.ListOptions{})
+	require.NoError(t, err, "Could not get list of nodes")
+	require.NotZero(t, len(nodes.Items), "No nodes found")
+
+	// Find the node that we spun up
+	for _, node := range nodes.Items {
+		for _, address := range node.Status.Addresses {
+			if address.Type == "ExternalIP" && address.Address == createdInstanceCreds.GetIPAddress() {
+				createdNode = &node
+				break
+			}
+		}
+		if createdNode != nil {
+			break
+		}
+	}
+	require.NotNil(t, createdNode, "Created node not found through Kubernetes API")
+
+	// Make sure the node is in a ready state
+	foundReady := false
+	for _, condition := range createdNode.Status.Conditions {
+		if condition.Type == v1.NodeReady {
```

Comment:
Nit: having a negation check with `continue` will make the code easier to read.

Reply:
Changed